core/src/main/scala/org/apache/spark/scheduler/ExecutorResourceInfo.scala

```diff
@@ -25,10 +25,13 @@ import org.apache.spark.resource.{ResourceAllocator, ResourceInformation}
  * information.
  * @param name Resource name
  * @param addresses Resource addresses provided by the executor
+ * @param numParts Number of ways each resource is subdivided when scheduling tasks
  */
-private[spark] class ExecutorResourceInfo(name: String, addresses: Seq[String])
+private[spark] class ExecutorResourceInfo(name: String, addresses: Seq[String],
+                                          numParts: Int)
   extends ResourceInformation(name, addresses.toArray) with ResourceAllocator {
 
   override protected def resourceName = this.name
   override protected def resourceAddresses = this.addresses
+  override protected def resourcesPerAddress = numParts
```

Comment:
I think we should rename. resourcesPerAddress could be confusing since an address is a single resource.  
I'm thinking we just call this tasksPerAddress and perhaps update this to be consistent everywhere - ie rename numParts as well.

Reply:
One issue here is that this would be part of the `ResourceAllocator` interface, of which there are two `WorkerResourceInfo`, which sets the `resourcesPerAddress` to 1, and `ExecutorResourceInfo`, which is the variable one. 

Could we all it something like `slotsPerAddress`?

Reply:
sure slotsPerAddress sounds good.